tests/framework/pkg/file/file.go

```diff
@@ -0,0 +1,68 @@
+package file
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+)
+
+// NewDir is a function that creates a directory with given directory name.
+// Ignores the returned error if the directory already exists.
+func NewDir(dirName string) (err error) {
+	err = os.Mkdir(dirName, 0777)
+	if err != nil {
+		if strings.Contains(err.Error(), "file exists") {
+			logrus.Infof("dir already exists: %v", err)
+			return nil
+		} else {
+			return
+		}
+	}
+
+	return
+}
+
+type FileName string
+
+// NewFile is a method that creates a file with the received file name.
+func (f FileName) NewFile(data []byte) (wdPath string, err error) {
+	err = os.WriteFile(string(f), data, 0644)
+	if err != nil {
+		return
```

Comment:
Should there be error handling similar to the NewDir() error handling? In case theres an error: if file already exists `return nil` else `return err`?

Reply:
No cause at this point we generate files, not a dir, that dir is an exception to prevent exits between different Jenkins stages. i.e. upgrade HA matrix, recurring runs, and now these RKE files, we want to see the errors on this